lib/BaseClient.ts

```diff
@@ -17,10 +18,10 @@ abstract class BaseClient {
     this.logger.info(`${this.constructor.name} status: ${ClientStatus[val]}`);
     this.status = val;
   }
-
-  public isDisabled(): boolean {
-    return this.status === ClientStatus.DISABLED;
+  public isConnected(): boolean {
+    return this.status === ClientStatus.CONNECTION_VERIFIED;
   }
+
```

Comment:
We can drop this blank line, but also here I'm thinking we should have `isDisconnected` and `isDisabled` methods as well. Or maybe it'd be cleaner to just have a expose the `status` field via a getter method, up to you.